src/extlib/logger.hpp

```diff
@@ -0,0 +1,88 @@
+#ifndef CPPLOGGER_HPP
+#define CPPLOGGER_HPP
+
+#include <chrono>
+#include <iostream>
+#include <string>
+#include <fstream>
+#include <filesystem>
+#include <sstream>
+#include <iomanip>
+
+enum class LogLevel {
+    INFO,
+    WARNING,
+    ERROR,
+    DEBUG
+};
+
+enum class LogOutput {
+    CONSOLE,
+    FILE
+};
```

Comment:
What will be the use case for this?
IMO, a log should always go into file anyway, only the console is optional. And what if we want to output to both file and console? This enum does not seem to solve that. For now, I think you can just log to file only. We can handle the console output later. So you can remove this enum.